File: Client/src/client/systems/garage.js
/// <reference types="@altv/types-client" />
/// <reference types="@altv/types-natives" />

import alt from 'alt-client';
import native from 'natives';


let garage;
let isWebViewOpen = false;
const interactionRadius = 2;
const pedFloatGarage = { x: 213.784, y: -808.47, z: 29.992 };

alt.on('keydown', (key) => {
    const playerPos = alt.Player.local.pos;
    const distance = playerPos.distanceTo(pedFloatGarage);

    if (distance <= interactionRadius) {
        if (key === "E".charCodeAt(0)) {
            if (!isWebViewOpen) {
                garage = new alt.WebView("http://resource/client/webview/garage/index.html");
                isWebViewOpen = true;
                garage.on('CloseGarageWebView', () => {
                    garage.destroy();
                    alt.showCursor(false);
                    alt.toggleGameControls(true);
                    garage = null;
                    isWebViewOpen = false;
                });
                garage.focus();
                alt.showCursor(true);
                alt.toggleGameControls(false);
            }
        }
    }
});


alt.onServer('Client:Garage:Open', (name, fuel, plate) => {
    if (isWebViewOpen && garage) {
        garage.emit('ShowGarage', name, fuel, plate);
    }
});

// Für den Backend aufruf des Events
alt.onServer('Client:Garage:CloseWebView', () => {
    if (isWebViewOpen && garage) {
        alt.showCursor(false);
        alt.toggleGameControls(true);
        garage.destroy();
        garage = null;
        isWebViewOpen = false;
    }
});

File: Los Angeles Life Client/Los Angeles Life Client/System/Garage.cs
using AltV.Net.Client;
using AltV.Net.Client.Elements.Data;
using AltV.Net.Client.Elements.Interfaces;
using AltV.Net.Data;

namespace Los_Angeles_Life_Client.System;

public abstract class Garage : Client
{
    private static bool _isWebViewOpen;
    private const int InteractionsRadius = 2;
    private static readonly Position PedFloatGarage = new((float)213.784, (float)-808.47, (float)29.992);
    private static IWebView? _garageHud;
    
    public static void Load()
    {
        Alt.OnKeyDown += (key) =>
        {
            var playerPos = Alt.LocalPlayer.Position;
            var distance = playerPos.Distance(PedFloatGarage);

            if (distance <= InteractionsRadius)
            {
                if (key == Key.E)
                {
                    //Alt.EmitServer("Client:Garage:SendPlayerInformation", 1000);
                    if (!_isWebViewOpen)
                    {
                        _garageHud = Alt.CreateWebView("http://resource/net6.0/webview/garage/index.html");
                        _isWebViewOpen = true;
                        _garageHud.On("CloseGarageWebView", () =>
                        {
                            _garageHud.Destroy();
                            Alt.ShowCursor(false);
                            Alt.GameControlsEnabled = true;
                            _garageHud = null;
                            _isWebViewOpen = false;
                        });

                        _garageHud.On("ParkIntoGarage", () =>
                        {
                            Alt.EmitServer("Client:Garage:ParkIntoGarage", 1000);
                        });
                        
                        _garageHud.Focus();
                        Alt.ShowCursor(true);
                        Alt.GameControlsEnabled = false;
                    }
                }
            }
        };
        
        Alt.OnServer("Client:Garage:Open", (string name, float fuel, string plate) =>
        {
            if (_garageHud != null && _isWebViewOpen)
            {
                _garageHud.Emit("ShowGarage", name, fuel, plate);
            }
        });
        
        Alt.OnServer("Client:Garage:CloseWebView", () =>
        {
            if (_garageHud != null && _isWebViewOpen )
            {
                Alt.ShowCursor(false);
                Alt.GameControlsEnabled = true;
                _garageHud.Destroy();
                _garageHud = null;
                _isWebViewOpen = false;
            }
        });
    }
}